usaspending_api/etl/management/commands/load_transactions_in_delta.py

```diff
@@ -0,0 +1,406 @@
+                    tidlu.detached_award_procurement_id = dap.detached_award_procurement_id
+                )
+                WHERE tidlu.detached_award_procurement_id IS NOT NULL AND dap.detached_award_procurement_id IS NULL
+                UNION ALL
+                SELECT id AS id_to_remove
+                FROM int.transaction_id_lookup AS tidlu LEFT JOIN raw.published_fabs AS pfabs ON (
+                    tidlu.published_fabs_id = pfabs.published_fabs_id
+                )
+                WHERE tidlu.published_fabs_id IS NOT NULL AND pfabs.published_fabs_id IS NULL
+            """
+        elif self.etl_level == "award_id_lookup":
+            id_col = "transaction_unique_id"
+            subquery = """
+                SELECT aidlu.transaction_unique_id AS id_to_remove
+                FROM int.award_id_lookup AS aidlu LEFT JOIN raw.detached_award_procurement AS dap ON (
+                    aidlu.detached_award_procurement_id = dap.detached_award_procurement_id
+                )
+                WHERE aidlu.detached_award_procurement_id IS NOT NULL AND dap.detached_award_procurement_id IS NULL
+                UNION ALL
+                SELECT aidlu.transaction_unique_id AS id_to_remove
+                FROM int.award_id_lookup AS aidlu LEFT JOIN raw.published_fabs AS pfabs ON (
+                    aidlu.published_fabs_id = pfabs.published_fabs_id
+                )
+                WHERE aidlu.published_fabs_id IS NOT NULL AND pfabs.published_fabs_id IS NULL
+            """
+
+        sql = f"""
+            MERGE INTO int.{self.etl_level}
+            USING (
+                {subquery}
+            ) AS deleted_records
+            ON {self.etl_level}.{id_col} = deleted_records.id_to_remove
+            WHEN MATCHED
+            THEN DELETE
+        """
+        return sql
+
+    def initial_run(self):
+        """
+        TODO:
+            - External Data values need to be added for all etl_levels (including transaction_ids)
+            - Initial value of the sequence should be set
+        """
+        #### transaction_id_lookup
+
+        # Capture start time of the transaction_id_lookup creation to update the "last_load_date" after completion
+        transaction_id_lookup_start_time = datetime.now(timezone.utc)
+
+        spark_s3_bucket = CONFIG.SPARK_S3_BUCKET
+        delta_lake_s3_bucket = CONFIG.DELTA_LAKE_S3_PATH
+        destination_database = "int"
+        destination_table = "transaction_id_lookup"
+
+        # Before creating table or running INSERT, make sure expected keys have no NULLs
+        self.logger.info("Checking for NULLs in key transaction columns")
+        tn_table_names = ("raw.transaction_normalized", "raw.transaction_normalized",
+                          "raw.detached_award_procurement", "raw.published_fabs")
+        tn_key_names = ("id", "transaction_unique_id", "detached_award_proc_unique", "afa_generated_unique")
```

Comment:
According to the current table schemas, you're right that I should check those 2 other columns for NULLs.  However, I also realized that the 2 columns I'm currently checking in `raw.transaction_normalized` are already set to `NOT NULL` in the Delta schema for `transaction_normalized`, as are the columns I'm checking later from `raw.awards`, so I can remove those 4 checks.  

This also raises the bigger question of whether the columns I'm checking here should be changed to be `NOT NULL` in the Delta schemas for `raw.published_fabs`, `raw.detached_award_procurement`, as well as `unique_award_key` in `transaction_normalized`.

Reply:
Slack comments:

Wil:
In general, we’ve avoided using too many constraints on the delta tables for performance and that enforcement should already have been taken care of at the Postgres level
[4:15](https://data-transparency-bfs.slack.com/archives/D042M8Q0Q8J/p1668460537336129)
Deciding if this should be a special case though because eventually those source tables won’t be done as a nuke/pave
New
[4:16](https://data-transparency-bfs.slack.com/archives/D042M8Q0Q8J/p1668460588642009)
I would say not for now at least because we’re still doing a full reload of all those tables, so Postgres’ enforcement of the constraints should still hold up

Me:
Based on your comments, should I double-check the constraints on the Postgres tables, and only check for NULLs in columns that allow for NULLs in the Postgres tables (if there are any)?

Wil:
Yes, I think this makes sense

Reply:
After looking at both the Delta and Postgres definitions of the fields in question in the tables in question, the only field that can be NULL in both Postgres and Delta is `unique_award_key` in `raw.transaction_normalized`, so changing code to only check that field in that table.